src/main/java/com/cleanroommc/groovyscript/compat/mods/tinkersconstruct/EntityMelting.java

```diff
@@ -0,0 +1,164 @@
+package com.cleanroommc.groovyscript.compat.mods.tinkersconstruct;
+
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.documentation.annotations.*;
+import com.cleanroommc.groovyscript.compat.mods.tinkersconstruct.recipe.EntityMeltingRecipe;
+import com.cleanroommc.groovyscript.core.mixin.tconstruct.TinkerRegistryAccessor;
+import com.cleanroommc.groovyscript.helper.SimpleObjectStream;
+import com.cleanroommc.groovyscript.helper.recipe.IRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import net.minecraft.entity.EntityList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fml.common.registry.EntityEntry;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RegistryDescription
+public class EntityMelting extends VirtualizedRegistry<EntityMeltingRecipe> {
+
+    @RecipeBuilderDescription(example = @Example(".fluidOutput(fluid('iron') * 500).input('minecraft','pig').register()"))

```

Comment:
i would suggest using the `resource` retriever to get the entity
```suggestion
    @RecipeBuilderDescription(example = @Example(".fluidOutput(fluid('iron') * 500).input(resource('minecraft:pig'))"))
```